namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+
+In other words, an expanded element is such an element that does not contain _Identifier_ (defined bellow) to any other elements but those defined in JSON or MSON namespaces.
+
+The expanded DOM MUST, however, keeps the track of what data structure was expanded and what from where.
+
+## Base Element
+
+In MSON, every data structure is a sub-type of another data structure, and, in transition, it is directly or indirectly derived from one of the MSON _Base Types_. This is expressed as an inheritance of elements in MSON DOM. Where the predecessor of an element is referred to as element's _Base Element_.
+
+Note not every MSON _Base Type_ is presented in JSON namespace primitive types and vice versa, see the table bellow:
+
+### JSON Namespace vs. MSON built-in types
+
+| JSON primitive |   JSON Namespace   | MSON Base Type |   MSON Namespace   |
+|:--------------:|:------------------:|:--------------:|:------------------:|
+|     boolean    |  JSON:Boolean Type |     boolean    |  MSON:Boolean Type |
+|     string     |  JSON:String Type  |     string     |  MSON:String Type  |
+|     number     |  JSON:Number Type  |     number     |  MSON:Number Type  |
+|      array     |   JSON:Array Type  |      array     |   MSON:Array Type  |
+|        -       |          -         |      enum      |   MSON:Enum Type   |
+|     object     |  JSON:Object Type  |     object     |  MSON:Object Type  |
+|      null      |   JSON:Null Type   |        -       |          -         |
+|        -       | JSON:Property Type |        -       | MSON:Property Type |
+
+
+# General Elements Definition
+
+General elements defined inside MSON namespaces but possibly reusable in another domain.
+
+## Identifier (Enum)
+
+Identifier of an element in the MSON namespace. The identifier MUST be either directly a `string` value or an `Id Element` element.
+
+### Members
+
+- (string)
+- (Id Element)
+
+## Id Element (Distinct Element)
+
+Element representing an identifier in the MSON namespace.
+
+### Properties
+
+- `element`: id (string, required, fixed)
+- `content` (string, required)
+
+## Distinct Element (Element)
+
+Abstract element representing an element that can be referenced or is a reference to another element.
+
+### Properties
+
+- `element` (Identifier, required) - identifier of this element's base element
+- `attributes`
+    - `id` (Identifier) - identifier of this element
+    - `ref` (Identifier) - reference to expanded element
+
+### Example
+
+```json
+{
+    "element": "number",
+    "attributes": {
+        "id": "my-number"
+    },
+    "content": 42
+}
+```
+
+which is identical to:
+
+```json
+{
+    "element": {
+        "element": "id",
+        "content": "number"
+    },
+    "attributes": {
+        "id": "my-number"
+    },
+    "content": 42
+}
+```
+
+and can be referred to as a _base element_:
+
+```json
+{
+    "element": "my-number"
```

Comment:
> . Seems to me there is a fundamental spec (around the namespacing one probably) that dictates referencing and re-use that Named Types should then leverage. 

There is none :( This acts instead of it.